### ISSUE TITLE
Improve error message when calling method with "previously freed instance" argument

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#35192]
**Godot version:**
3.2 master (3af0400a)

**OS/device including version:**
Any

**Issue description:**
When calling a GDScript method with a "previously freed instance" parameter, the error which is raised is not as clear as it could be. Maybe also for the "null instance" case, though I don't know how to reproduce it.

```
var node = Node.new()
node.free()
add_child(node)
```

raises:
```
Invalid type in function 'add_child' in base 'Node2D (Node2D.gd)'. The Object-derived class of argument 1 (previously freed instance) is not a subclass of the expected argument class.
```

Relevant code to adjust:
https://github.com/godotengine/godot/blob/3af0400a32a2f81aad0b56940af0d6a92b6ffc92/modules/gdscript/gdscript_function.cpp#L166-L176

**Steps to reproduce:**
- Use above snippet on any Node-derived script.

**Minimal reproduction project:**
See above.

**Comment by timothyqiu** [comment]
I'm a bit confused when reading the documentation for [Object.free()](http://docs.godotengine.org/en/latest/classes/class_object.html#class-object-method-free)

> Deletes the object from memory. Any pre-existing reference to the freed object will now return `null`.

Isn't `node` a reference to the object? If yes, then it should become `null` after `node.free()` I think.

Or, maybe the documentation is referring to the Reference class?

**Comment by bojidar-bg** [comment]
Well, there is the thing.. pre-existing references become "previously freed instance" instead of "null".
Note that things can get much worse from there on as of now:
```gdscript
var node = Node.new()
node.free()
var node2 = Node.new()
add_child(node) # No error!
print(node, node2) # e.g. [Node:1370][Node:1370]
```
But:
```gdscript
var node = Node.new()
node.free()
var node2 = Sprite.new()
add_child(node) # Error: Same as before (if you are lucky with malloc)
```

**Comment by remram44** [comment]
@bojidar-bg I *think* this is what was fixed by #38119

**Comment by akien-mga** [comment]
This is fixed indeed.
